hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,274 @@
+
+        Parameters
+        ----------
+        sb_position : tuple
+            The sideband position (y, x), referred to the non-shifted FFT.
+
+        Returns
+        -------
+        Sideband size (y, x), referred to the unshifted FFT.
+        """
+        if isinstance(sb_position, BaseSignal):
+            sb_position = sb_position.data
+
+        if self.axes_manager.navigation_size:
+            sb_size = np.zeros(self.axes_manager.navigation_size)
+
+            for i in range(self.axes_manager.navigation_size):
+                sb_size[i] = find_sideband_size(self.axes_manager.signal_shape, sb_position[i])
+
+        else:
+            sb_size = find_sideband_size(self.axes_manager.signal_shape, sb_position)
+
+        return sb_size
+
+    def reconstruct_phase(self, reference=None, sb_size=None, sb_smooth=None, sb_unit=None,
+                          sb='lower', sb_position=None, output_shape=None, plotting=False):
+        """Reconstruct electron holograms.
+
+        Parameters
+        ----------
+        reference : ndarray, :class:`~hyperspy.signals.Signal2D
+            Vacuum reference hologram.
+        sb_size : float
+            Sideband radius of the aperture in corresponding unit (see 'sb_unit'). If None,
+            the radius of the aperture is set to 1/3 of the distance between sideband and
+            centerband.
+        sb_smooth : float, optional
+            Smoothness of the aperture in the same unit as sb_size.
+        sb_unit : str, optional
+            Unit of the two sideband parameters 'sb_size' and 'sb_smoothness'.
+            Default: None - Sideband size given in pixels
+            'nm': Size and smoothness of the aperture are given in 1/nm.
+            'mrad': Size and smoothness of the aperture are given in mrad.
+        sb : str, optional
+            Select which sideband is selected. 'upper' or 'lower'.
+        sb_position : tuple, optional
+            Sideband position in pixel. If None, sideband is determined automatically from FFT.
+        output_shape: tuple, optional
+            Choose a new output shape. Default is the shape of the input hologram. The output
+            shape should not be larger than the input shape.
+        plotting : boolean
+            Shows details of the reconstruction (i.e. SB selection).
+
+        Returns
+        -------
+        wave : :class:`~hyperspy.signals.WaveImage
+            Reconstructed electron wave. By default object wave is devided by reference wave
+
+        Notes
+        -----
+        Use wave.rec_param to extract reconstruction parameters, which can be used for batch
+        processing.
+        """
+
+        # Unfolding self:
+
+        # Parsing reference:
+        if not isinstance(reference, HologramImage):
+            if isinstance(reference, Signal2D):
+                warnings.warn('The reference image signal type is not HologramImage. It will be converted to '
```

Comment:
Chenged it to logger:
```python
_logger.warning('The reference image signal type is not HologramImage. It will '
                                'be converted to HologramImage automatically.')
```